dvc/remote/gdrive.py

```diff
@@ -92,25 +93,27 @@ def gdrive_upload_file(
         item = self.drive.CreateFile(
             {"title": args["title"], "parents": [{"id": args["parent_id"]}]}
         )
-        self.upload_file(item, no_progress_bar, from_file, progress_name)
-        return item
 
-    def upload_file(self, item, no_progress_bar, from_file, progress_name):
-        with open(from_file, "rb") as opened_file:
-            if not no_progress_bar:
-                opened_file = TrackFileReadProgress(progress_name, opened_file)
-            # PyDrive doesn't like content property setting for empty files
-            # https://github.com/gsuitedevs/PyDrive/issues/121
-            if os.stat(from_file).st_size:
-                item.content = opened_file
-            item.Upload()
+        with open(from_file, "rb") as fobj:
+            total = os.path.getsize(from_file)
+            with Tqdm.wrapattr(
+                fobj,
+                "read",
+                desc=progress_name,
+                total=total,
+                disable=no_progress_bar,
+            ) as wrapped:
+                # PyDrive doesn't like content property setting for empty files
+                # https://github.com/gsuitedevs/PyDrive/issues/121
+                if total:
+                    item.content = wrapped
+                item.Upload()
+        return item
 
     @gdrive_retry
     def gdrive_download_file(
         self, file_id, to_file, progress_name, no_progress_bar
     ):
-        from dvc.progress import Tqdm
-
         gdrive_file = self.drive.CreateFile({"id": file_id})
         with Tqdm(
```

Comment:
maybe we should remove this completely for now... Right now, a user will see somwthing like "0/10GB" for 10mins before suddenly the operation completes and the bar disappears. It's probably better if instead we say "Downloading...".

Reply:
@casperdcl Let's keep it as is, at least there are some notifications (binary, but still) 🙂

Reply:
yes but if I saw a "0/10GB" doing nothing for a minute I'd assume something was wrong and nothing was happening

Reply:
@casperdcl But what our options at this point? I suppose:
1) remove it (which will just leave an empty screen, which is worse)
2) add some kind of temporary spinner to keep the user entertained
3) dive into pydrive and hack a proper solution for us and submit it to pydrive
am I missing something? If not, which one do you suggest? 🙂

Reply:
Definitely option 2 (well, not a spinner, but just the word "Downloading...")

1. agreed bad
2. technically correct
3. would take a while and better to do in other PR(s)
4. leave as-is (0/10GB hanging); I think **worse** than (1) because I'd be **more** likely to incorrectly think the program was hanging

Reply:
@casperdcl The word will be static? If so, it doesn't seem very different from the current approach. Maybe `Downloading... (no progress available)` to explain? 🙂 Though, we've just merged gdrive support and haven't even documented it yet, so perhaps we are overthinking the importance of this at this particular moment and we probably could simply wait for that stuff to get properly fixed before publically announcing the support.

Reply:
@shcheklein my 2cs - I would vote for spinner in these cases. And I think it's good to have them in our toolkit anyways - will give us more flexibility in terms of building UI. @casperdcl do you have a spinner/super-compact mode for TQDM ? :)

Reply:
yeah idk I still prefer a correct static word to a misleading and incorrect 0/10GB. Of course we could try to improve the static word but atm `tqdm` doesn't support threaded spinners (i.e. a background thread updates the screen even though no calls to `update` or `__iter__` were performed)

Reply:
Always happy to accept a PR :)

Reply:
Guys, we are overthinking this one, it will probably get fixed properly very soon, so any placeholder will do (even the previous one, we used to have it for gs for the longest time and didn't receive that much complains, so I'm not really worried about the current temp solution). I'm fine with the current static "Downloading" for now.